langchain/memory/chat_memory.py

```diff
@@ -0,0 +1,44 @@
+from typing import Any, Dict, List, Optional
+
+from pydantic import BaseModel, Field
+
+from langchain.memory.utils import get_prompt_input_key
+from langchain.schema import AIMessage, BaseMessage, HumanMessage, Memory
+
+
+class ChatMessageHistory(BaseModel):
+    messages: List[BaseMessage] = Field(default_factory=list)
+
+    def add_user_message(self, message: str) -> None:
+        self.messages.append(HumanMessage(content=message))
+
+    def add_ai_message(self, message: str) -> None:
+        self.messages.append(AIMessage(content=message))
+
+    def clear(self) -> None:
+        self.messages = []
+
+
+class ChatMemoryMixin(Memory):
```

Comment:
this isn't really a mixing -- it's a base class. Rename to `BaseChatMemory`? and inherit ABC